framework/core/js/src/common/extenders/Routes.ts

```diff
@@ -0,0 +1,36 @@
+import Application, { FlarumGenericRoute } from '../Application';
+import ExtenderInterface, { ExtensionModuleInterface } from './ExtenderInterface';
+
+type HelperRoute = (...args: any) => string;
+
+export default class Routes implements ExtenderInterface {
+  private routes: Record<string, FlarumGenericRoute> = {};
+  private helpers: Record<string, HelperRoute> = {};
+
+  /**
+   * Add a mithril route to the application.
+   *
+   * @param name The name of the route.
+   * @param path The path of the route.
+   * @param component must extend `Page` component.
+   */
+  add(name: string, path: `/${string}`, component: any): Routes {
+    this.routes[name] = { path, component };
+
+    return this;
+  }
+
+  helper(name: string, callback: HelperRoute): Routes {
+    this.helpers[name] = callback;
+
+    return this;
+  }
+
+  extend(app: Application, extension: ExtensionModuleInterface) {
+    Object.assign(app.routes, this.routes);
+
+    if (Object.keys(this.helpers).length) {
+      Object.assign(app.route, this.helpers);
+    }
```

Comment:
Why is this conditional? Assigning in an empty object should be a no-op, so why not do it regardless?

Reply:
I honestly don't recall why I specifically did this here and not elsewhere, it is unnecessary.